index.js

```diff
@@ -74,6 +74,20 @@ function parseIntoPreview(data) {
   });
 }
 
+function parseIntoTrackArray(data){
+  if(!data.name)
+    new Error("Data doesn't seem to be of the right shape to parse")
+  if(!data.tracks)
+    return Promise.resolve([data])//track
+  if(data.tracks.items)
+      if(data.tracks.items[0].track)
+          return Promise.resolve(data.tracks.items.map(t => t.track));//playlist
+      else
+          return Promise.resolve(data.tracks.items)//album
+  else(data.tracks)
```

Comment:
was this supposed to be an `else if`?

Reply:
if it was, and we change the first one to `data.type === "track"` we will still have a last else case to cover, which could probably be throwing an error like "could not get any tracks"

Reply:
> was this supposed to be an `else if`?

hmm yes interesting